app/models/mhv_account.rb

```diff
@@ -148,7 +153,9 @@ def previously_registered?
 
   def setup
     raise StandardError, 'You must use find_or_initialize_by(user_uuid: #)' if user_uuid.nil?
-    check_eligibility
-    check_terms_acceptance if may_check_terms_acceptance?
+    if beta_enabled?(user_uuid)
+      check_eligibility
+      check_terms_acceptance if may_check_terms_acceptance?
+    end
```

Comment:
Should we add an else condition here to ensure that the account state is unknown? It should be, since it's not persisted, but for the sake of readability and a sense of what the expectation ought to be in the user_serializer, this might be helpful.

but of course we don't have an event for setting it to unknown, so it might be tricky to do.

NITPICK.